pkg/util/cert_pool.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2022 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+)
+
+// CertPool is a set of certificates.
+type CertPool struct {
```

Comment:
Ref. https://github.com/cert-manager/trust-manager/pull/184#discussion_r1335625480 I think this type should be defined in an internal package. Or as an unexported type in the package it's used. We mustn't tempt anyone to use it. :-)

Reply:
Sounds good!
Let's do it!

Reply:
`CertPool` and its methods to unreported ones, so `certPool` is used by `util` package only